### PR TITLE
Upgraded to latest NSB

### PR DESCRIPTION
**Pull request by SzymonPobiega**
Connects to https://github.com/Particular/PlatformDevelopment/issues/706

This is blocking the private beta as we can't create a beta release without first upgrading to latest unstable. This PR does not have any groundbraking changes, only:
 * Replacing usages of `IConfigureComponents` with `ResolveAll`
 * Upgrading ATTs
 * Replacing `IWantToRun...` with a feature in the performance test app

### CODE REVIEW

**Comment by SzymonPobiega** [comment]
@Particular/persistence-maintainers please review ASAP.

**Comment by andreasohlund** [comment]
Shouldn't we just remove the perf test app?

**Comment by andreasohlund** [comment]
I assume we need the resolve all trick in RavenDB as well?

Context: https://github.com/Particular/docs.particular.net/pull/1389

**Comment by DavidBoike** [comment]
@andreasohlund:

> I assume we need the resolve all trick in RavenDB as well?

What?

@SzymonPobiega: This is impossible to review on GitHub but I could clone. Do we actually run the perf tests in TeamCity?

**Comment by SzymonPobiega** [comment]
@DavidBoike From what I see we do, but they are failing silently (and the build is green). I did not feel like fixing/removing this project should be in scope of this PR

**Comment by SzymonPobiega** [comment]
@DavidBoike I've split it into two commits so now you should be able to review https://github.com/Particular/NServiceBus.NHibernate/pull/181/commits/a03b0a6772099ed1ba921cf0fab85aeed8d45da4

**Comment by SzymonPobiega** [comment]
@DavidBoike regarding trick, @andreasohlund meant that the "hack" used to allow installers to work (via `IConfigureComponents`) would break after upgrading RavenDB to latest NSB core and that RavenDB needs a PR similar to this one to fix it.

File: src/NServiceBus.NHibernate.PerformanceTests/Program.cs [hunk 1]
```diff
@@ -75,8 +76,8 @@ static void Main(string[] args)
             {
                 config.EnableOutbox();
             }
-
-            config.RunWhenEndpointStartsAndStops(new Loader(async session =>
+            config.EnableFeature<LoaderFeture>();
```
Comment (colin-higgins) [inline src/NServiceBus.NHibernate.PerformanceTests/Program.cs]: LoaderFeture typo

LeaderFeature

Comment (SzymonPobiega) [inline src/NServiceBus.NHibernate.PerformanceTests/Program.cs]: Fixed

**Comment by HEskandari** [comment]
@SzymonPobiega  a03b0a6 Looks good to me.

File: src/NServiceBus.NHibernate.PerformanceTests/NServiceBus.NHibernate.PerformanceTests.csproj [hunk 2]
```diff
@@ -50,7 +50,7 @@
       <Private>True</Private>
     </Reference>
     <Reference Include="NServiceBus.Core, Version=6.0.0.0, Culture=neutral, PublicKeyToken=9fc386479f8a226c, processorArchitecture=MSIL">
-      <HintPath>..\packages\NServiceBus.6.0.0-unstable1616\lib\net452\NServiceBus.Core.dll</HintPath>
+      <HintPath>..\packages\NServiceBus.6.0.0-unstable1786\lib\net452\NServiceBus.Core.dll</HintPath>
```
Comment (danielmarbach) [inline src/NServiceBus.NHibernate.PerformanceTests/NServiceBus.NHibernate.PerformanceTests.csproj]: Housten we have a problem. @SimonCropp unlisted those unstables.

**Comment by danielmarbach** [comment]
@HEskandari we cannot merge this since the unstable referenced is no longer available :( This would break develop build. We need to rebase this one to an alpha build or reupload this specific unstable version. Thoughts @SimonCropp ?

File: src/NServiceBus.NHibernate.PerformanceTests/NServiceBus.NHibernate.PerformanceTests.csproj (hunk 2 above)
Comment (SimonCropp) [inline src/NServiceBus.NHibernate.PerformanceTests/NServiceBus.NHibernate.PerformanceTests.csproj]: yep. update to the beta

**Comment by danielmarbach** [comment]
Made it WIP because it cannot be merged until we resolved the issue with the missing package. Apart from that the PR is good

**Comment by danielmarbach** [comment]
I'll take care of it

**Comment by danielmarbach** [comment]
@SzymonPobiega force pushed as discussed. @HEskandari can you merge when green?

**Comment by danielmarbach** [comment]
Can we get this in?

**Comment by andreasohlund** [comment]
Looks good to me

**Comment by SzymonPobiega** [comment]
@danielmarbach :dancer:

**Pull request merged**